cardano-api/src/Cardano/Api/Query.hs

```diff
@@ -110,6 +117,20 @@ data QueryInMode mode result where
     :: ConsensusModeIsMultiEra mode
     -> QueryInMode mode (EraHistory mode)
 
+  QuerySystemStart
+    :: ConsensusModeIsMultiEra mode
+    -> QueryInMode mode SystemStart
+
+data EraHistory mode where
+  EraHistory
+    :: ConsensusBlockForMode mode ~ Consensus.HardForkBlock xs
+    => ConsensusMode mode
+    -> History.Interpreter xs
+    -> EraHistory mode
+
+getProgress :: SlotNo -> EraHistory mode -> Either Qry.PastHorizonException (RelativeTime, SlotLength)
+getProgress slotNo (EraHistory _ interpreter) = Qry.interpretQuery interpreter (Qry.slotToWallclock slotNo)
```

Comment:
Mmm. I suppose we'll need to provide some re-export or wrapping of the API for the `EraHistory` to make it useful.

Reply:
What's the action I need to take here?